designer/server/src/main/scala/pl/touk/nussknacker/ui/definition/ScenarioPropertiesConfigFinalizer.scala

```diff
@@ -0,0 +1,18 @@
+package pl.touk.nussknacker.ui.definition
+
+import cats.implicits.catsSyntaxSemigroup
+import pl.touk.nussknacker.engine.api.component.{AdditionalUIConfigProvider, ScenarioPropertyConfig}
+import pl.touk.nussknacker.engine.api.process.ProcessingType
+
+class ScenarioPropertiesConfigFinalizer(
+    additionalUIConfigProvider: AdditionalUIConfigProvider,
+) {
+
+  def finalizeScenarioProperties(
+      scenarioPropertiesConfig: Map[String, ScenarioPropertyConfig],
+      processingType: ProcessingType
```

Comment:
Can you add a `processingType` into `ProcessingTypeData`? Thanks to that we can move this from parameter's list to the constrctor.